front/src/statistics/Statistics.js

```diff
@@ -0,0 +1,310 @@
+import React, { useState, useEffect } from 'react';
+import Chart from 'react-apexcharts';
+import '../App.css';
+import './Statistics.css';
+
+// Date management
+import MomentUtils from '@date-io/moment';
+import moment from 'moment';
+import 'moment/locale/fi';
+
+// Material UI components
+import Grid from '@material-ui/core/Grid';
+
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker,
+} from '@material-ui/pickers';
+
+import axios from 'axios';
+
+// Translation
+import data from '../texts/texts.json';
+
+let lang = 'fi';
+if (localStorage.getItem('language') === '0') {
+  lang = 'fi';
+} else if (localStorage.getItem('language') === '1') {
+  lang = 'en';
+}
+
+moment.locale(lang);
+const { statistics } = data;
+const fin = localStorage.getItem('language');
+
+const Statistics = () => {
+  const [date, setDate] = useState(new Date());
+
+  const [monthlyUsers, setMonthlyUsers] = useState([]);
+  const [monthOptions, setMonthOptions] = useState({});
+  const [monthSeries, setMonthSeries] = useState([]);
+
+  const [dayOptions, setDayOptions] = useState({});
+  const [daySeries, setDaySeries] = useState([]);
+  const [dailyUsers, setDailyUsers] = useState([]);
+
+  const [dayNumber, setDayNumber] = useState([]);
+
+  useEffect(() => {
+    const firstDate = moment(date).startOf('month').format('YYYY-MM-DD');
+    const lastDate = moment(date).endOf('month').format('YYYY-MM-DD');
+    const singleDay = moment(date).format('YYYY-MM-DD');
+
+    // dayNum is used as an index in monthlyUsers[] to define the total visitors of a day
+    const dayNumberSplit = singleDay.split('-');
+    let dayNum = dayNumberSplit[2];
+    dayNum = dayNum.startsWith('0') ? dayNum.substring(1) : dayNum;
+    setDayNumber(dayNum);
+
+    const fetchData = async () => {
+      const monthResult = await getMonthlyVisitors(firstDate, lastDate); // eslint-disable-line
+      const dayResult = await getDailyVisitors(singleDay); // eslint-disable-line
+      setDailyUsers(dayResult);
+      setMonthlyUsers(monthResult);
+    };
+    fetchData();
+  }, [date]);
+
+  useEffect(() => {
+    // Count monthly visitors
+    let day = 1;
+    let dayArray = [];
+    monthlyUsers.forEach((user) => { // eslint-disable-line
+      dayArray = dayArray.concat(day);
+      day += 1;
```

Comment:
If I understood right you want an array that looks something like this:
[1, 2, ..., i], where i is the number of users

you can do this easier by:
Array.from({length: monthlyUsers.length()}, (_, i) => i + 1)

It creates an array of [0-monthlyUsers.length()], then increments everything by one.